db/migrate/20190724191053_update_itunes_categories.rb

```diff
@@ -0,0 +1,187 @@
+
+      # 'Science & Medicine' sub 'Medicine' -> 'Health & Fitness' sub 'Medicine'
+      if category.name == 'Science & Medicine' && category.subcategories.include?('Medicine')
+        category.name = 'Health & Fitness'
+        category.subcategories = ['Medicine']
+      end
+
+      # 'Science & Medicine' -> 'Science'
+      if category.name == 'Science & Medicine'
+        say(category.to_s)
+        category.name = 'Science'
+      end
+
+      # 'Society & Culture' sub 'History' top-level 'History' no subs
+      if category.name == 'Society & Culture' && category.subcategories.include?('History')
+        category.name = 'History'
+        category.subcategories = []
+      end
+
+      # 'Sports & Recreation' sub 'Amateur' discontinued
+      if category.name == 'Sports & Recreation' && category.subcategories.include?('Amateur')
+        category.subcategories.delete_at(category.subcategories.index('Amateur'))
+      end
+
+      # 'Sports & Recreation' sub 'College & High School' discontinued
+      if category.name == 'Sports & Recreation' && category.subcategories.include?('College & High School')
+        category.subcategories.delete_at(category.subcategories.index('College & High School'))
+      end
+
+      # 'Sports & Recreation' sub 'Outdoor' discontinued
+      if category.name == 'Sports & Recreation' && category.subcategories.include?('Outdoor')
+        category.subcategories.delete_at(category.subcategories.index('Outdoor'))
+      end
+
+      # 'Sports & Recreation' sub 'Professional' discontinued
+      if category.name == 'Sports & Recreation' && category.subcategories.include?('Professional')
+        category.subcategories.delete_at(category.subcategories.index('Professional'))
+      end
+
+      # 'Sports & Recreation' -> 'Sports'
+      if category.name == 'Sports & Recreation'
+        category.name = 'Sports'
+      end
+
+      # 'Technology' sub 'Gadgets' discontinued
+      if category.name == 'Technology' && category.subcategories.include?('Gadgets')
+        category.subcategories.delete_at(category.subcategories.index('Gadgets'))
+      end
+
+      # 'Technology' sub 'Tech News' -> 'News' sub 'Tech News'
+      if category.name == 'Technology' && category.subcategories.include?('Tech News')
+        category.name = 'News'
+        category.subcategories = ['Tech News']
+      end
+
+      # 'Technology' sub 'Podcasting' discontinued
+      if category.name == 'Technology' && category.subcategories.include?('Podcasting')
+        category.subcategories.delete_at(category.subcategories.index('Podcasting'))
+      end
+
+      # 'Technology' sub 'Software How-To' discontinued
+      if category.name == 'Technology' && category.subcategories.include?('Software How-To')
+        category.subcategories.delete_at(category.subcategories.index('Software How-To'))
+      end
+
+      if category.changed?
+        puts "updated category[#{category.id}] for podcast[#{category.podcast}] => #{category.changes}"
```

Comment:
Log outputs nothing for `category.podcast`. Not very useful for communicating to support which podcasts are being updated.

Reply:
Hmm... it's possible we have categories hanging around for deleted podcast_ids.  Or your dev environment is missing the `podcasts` table?

Either way - logging `#{category.podcast_id}` will just log the foreign key out of the database, instead of joining the podcasts table at all.  And is probably enough info for our case?

Reply:
Yes, I actually found later that I am missing Podcast data. 💯 good guess. I already got Paloma this information and she is reviewing it, so we can probably look at this logging as needed.